src/web/services/deadline-extension-helper.ts

```diff
@@ -0,0 +1,114 @@
+import {
```

Comment:
Some feedback is provided in the main PR

Reply:
I'll wait till the front end finishes to rebase this over

Reply:
Actually my bad, I can just rebase now. I'll need to do it to see the diffs anyway